src/Assert.php

```diff
@@ -399,7 +405,44 @@ public static function isInstanceOfAny($value, array $classes, $message = '')
         }
 
         static::reportInvalidArgument(sprintf(
-            $message ?: 'Expected an instance of any of %2$s. Got: %s',
+            $message ?: 'Expected an instance of this class or to this class among his parents of any of %2$s. Got: %s',
+            static::typeToString($value),
+            implode(', ', array_map(array('static', 'valueToString'), $classes))
+        ));
+    }
+
+    public static function isAOf($value, $class, $message = '')
+    {
+        if (!is_string($class) || !is_a($value, $class, is_string($value))) {
+            static::reportInvalidArgument(sprintf(
+                $message ?: 'Expected an instance of this class or to this class among his parents %2$s. Got: %s',
+                static::typeToString($value),
+                $class
+            ));
+        }
+    }
+
+    public static function notAOf($value, $class, $message = '')
+    {
+        if (!is_string($class) || is_a($value, $class, is_string($value))) {
+            static::reportInvalidArgument(sprintf(
+                $message ?: 'Expected an instance of this class or to this class among his parents other than %2$s. Got: %s',
+                static::typeToString($value),
+                $class
+            ));
+        }
+    }
+
+    public static function isAOfAny($value, array $classes, $message = '')
+    {
+        foreach ($classes as $class) {
+            if (!is_string($class) || is_a($value, $class, is_string($value))) {
```

Comment:
This statement must be changed to `if (is_string($class) && is_a($value, $class, is_string($value))) {` otherwise this test case `array('isAOfAny', array('ArrayIterator', array(123)), true),` would be passed. The correct test case would be `array('isAOfAny', array('ArrayIterator', array(123)), false),`